app.ts

```diff
@@ -0,0 +1,81 @@
+interface Student {
+    address?: string;
+    age?: number;
+    avgGrade?: number;
+    city?: string;
+    email: string;
+    id?: number;
+    firstName: string;
+    lastName: string;
+  }
+
+//  ex 1 
+  function fetchStudentData():Partial<Student>{
+      return {
+        avgGrade: 20,
+        id: 12345,
+        firstName: "Tasos",
+        lastName: "Matatakis"   
+      }
+    }
+
+
+// ex 2
+function sendStudentData(): Required<Student> {
+    return {
+      address: "Neverland 2",
+      age: 50,
+      avgGrade: 20,
+      city: "Atlantis",
+      email: "nobody@gmail.com",
+      id: 12345,
+      firstName: "Tasos",
+      lastName: "Matatakis"
+    };
+  }
+
+
```

Comment:
Nice implementation of Required but the function does not update the data in the server